src/hooks/useTimeout/index.test.js

```diff
@@ -0,0 +1,72 @@
+import { useEffect } from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import useTimeout from '.';
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor } from '@testing-library/react';
+
+describe('useTimeout hook', () => {
+  const delay = 1000;
+  let callback;
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+    callback = jest.fn();
+  });
+
+  it('should execute callback function after 1 second', async () => {
+    // Act
+    const { unmount } = renderHook(() => {
+      const createTimeout = useTimeout();
+
+      useEffect(() => {
+        createTimeout(callback, delay);
+      }, []);
+    });
+
+    // Assert
+    jest.advanceTimersByTime(delay);
+    await waitFor(() => expect(callback).toHaveBeenCalledTimes(1));
+
+    expect(clearTimeout).toHaveBeenCalledTimes(0);
+    unmount();
+    expect(clearTimeout).toHaveBeenCalledTimes(1);
+  });
+
+  it('should execute the callback functions after 1 second and 1.5 second', async () => {
+    // Arrange
+    const callback2 = jest.fn();
+    const delay2 = 1500;
+
+    // Act
+    const { unmount } = renderHook(() => {
+      const createTimeout = useTimeout();
+
+      useEffect(() => {
+        createTimeout(callback, delay);
+        createTimeout(callback2, delay2);
+      }, []);
```

Comment:
`useEffect` is not necessary here right?

```suggestion
      createTimeout(callback, delay);
      createTimeout(callback2, delay2);
```

I mean, `useTimeout` already has the `useEffect` inside, and in our real scenario, we are not using `useTimeout` inside `useEffect`.

Reply:
I use the useEffect in renderHook's to be able to execute the **createTimemout** function. it is the easiest way to use it.
If I don't put it, it will show me that the setTimeout has been called too many times. 

![image](https://user-images.githubusercontent.com/84402180/122561105-0b6d6800-d007-11eb-93e0-67c5a383c730.png)

what I can do is simulate that these calls to **createTimeout** are made within a function, and trigger the function by clicking on a fake button.

Reply:
fixed!